shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/check/consistency/DataConsistencyCheckerImpl.java

```diff
@@ -119,31 +126,49 @@ private long count(final DataSource dataSource, final String table, final Databa
                 jobContext.getJobConfig().getRuleConfig().getTarget().getType(), jobContext.getJobConfig().getRuleConfig().getTarget().getParameter());
         checkDatabaseTypeSupportedOrNot(supportedDatabaseTypes, targetConfig.getDatabaseType().getName());
         Collection<String> logicTableNames = jobContext.getTaskConfigs().stream().flatMap(each -> each.getDumperConfig().getTableNameMap().values().stream()).distinct().collect(Collectors.toList());
-        Map<String, Collection<String>> tablesColumnNamesMap = getTablesColumnNamesMap(sourceConfig);
+        Map<String, TableMetaData> tableMetaDataMap = getTablesColumnsMap(sourceConfig, logicTableNames);
         logicTableNames.forEach(each -> {
             //TODO put to preparer
-            if (!tablesColumnNamesMap.containsKey(each)) {
+            if (!tableMetaDataMap.containsKey(each)) {
                 throw new DataCheckFailException(String.format("could not get table columns for '%s'", each));
```

Comment:
not sure why upcon code like 
```
DataCheckFailException("count check failed, table=" + table, ex)
```

but here is String.format ?

Reply:
OK, I'll update the exception message style next time.